Update docs.md. Added module hierarchy paragraph. (#22919)
vlang/v: doc/docs.md

current_os := user_os()
println('Your OS is ${current_os}.')
```
### Module hierarchy

> [!NOTE]
> This section is valid when .v files are not in the project's root directory.

Modules names in .v files, must match the name of their directory.
 

> A .v file `./abc/source.v`  must start with `module abc`. All other .v files in this directory 

Reviewer: ```suggestion
A .v file `./abc/source.v` must start with `module abc`. All .v files in this directory 
```

belong to the same module abc and should start with `module abc`.

> If you have `abc/def/`, with .v files in both folders you can `import abc`, but you will have 

Reviewer: ```suggestion
If you have `abc/def/`, and .v files in both folders, you can `import abc`, but you will have 
```

to `import abc.def` too, to get to the symbols in the subfolder.

In `module name` statement, name never repeats directory's hierarchy, but only its directory.

> So in `abc/def/source.v`  first line will be `module def`, and never `module abc.def`.

Reviewer: ```suggestion
So in `abc/def/source.v` the first line will be `module def`, and not `module abc.def`.
```

> `import module_name` statements must respect file hierarchy, also you cannot import def, only

Reviewer: ```suggestion
`import module_name` statements must respect file hierarchy, you cannot `import def`, only
```

> abc.def

Reviewer: ```suggestion
`abc.def`
```

> Refering to a module symbol such function or variable only requires module name as prefix:

Reviewer: ```suggestion
Refering to a module symbol such as a function or const, only needs module name as prefix:
```

```v ignore
module def

// func is a dummy example function.
pub fn func() {
	println('func')
}
```

> will be called:

Reviewer: ```suggestion
can be called like this:
```

```v ignore
module main

import def

fn main() {
	def.func()
}
```

> A function, located in `abc/def/source.v`,  is called with `def.func()`, not `abc.def.func()`

Reviewer: ```suggestion
A function, located in `abc/def/source.v`, is called with `def.func()`, not `abc.def.func()`
```

> This always implies a single prefix, whatever sub-module depth. This behavior flattens 

Reviewer: ```suggestion
This always implies a *single prefix*, whatever sub-module depth. This behavior flattens 
```

> modules/sub-modules hierarchy. Should you have two modules with the same name in differents

Reviewer: ```suggestion
modules/sub-modules hierarchy. Should you have two modules with the same name in different
```

> directories then you should use Module import aliasing.

Reviewer: ```suggestion
directories, then you should use Module import aliasing (see below).
```
